src/base/dof_map_constraints.C

```diff
@@ -2255,8 +2255,8 @@ void DofMap::enforce_constraints_on_jacobian (const NonlinearImplicitSystem & sy
       const DofConstraintRow constraint_row = pr.second;
 
       for (const auto & j : constraint_row)
-        jac->set(constrained_dof, j.first, j.second);
-      jac->set(constrained_dof, constrained_dof, -1);
```

Comment:
Wait, this wasn't just a "didn't notice how signs get flipped and flipped again in some weird way" issue, it was an explicitly set -1 on the diagonal?  How'd we decide on that?

Reply:
I would like to be able to say that I put careful thought into this and somehow my meticulous thought went astray...but I think the truth is that this code was rather thoughtless...